src/integration-test/java/com/commercetools/sync/integration/commons/utils/ITUtils.java

```diff
@@ -65,21 +64,30 @@ public static String getStatisticsAsJSONString(@Nonnull final BaseSyncStatistics
      * @param resourceMapper       defines a mapper function that should be applied on each resource in the fetched page
      *                             from the query on the specified CTP project.
      */
-    public static <T extends Resource, C extends QueryDsl<T, C>> void queryAndApply(
+    public static <T extends Resource, C extends QueryDsl<T, C>> void queryAndExecute(
         @Nonnull final SphereClient ctpClient,
         @Nonnull final Supplier<QueryDsl<T, C>> queryRequestSupplier,
         @Nonnull final Function<T, SphereRequest<T>> resourceMapper) {
+        queryAll(ctpClient, queryRequestSupplier.get(), resourceMapper)
+            .thenApply(sphereRequests -> sphereRequests.stream()
+                                                       .map(ctpClient::execute)
+                                                       .map(CompletionStage::toCompletableFuture))
+            .thenCompose(ITUtils::toAllOf)
+            .toCompletableFuture().join();
+    }
+
 
-        final Function<List<T>, Stream<CompletableFuture<T>>> pageMapper =
-            pageElements -> pageElements.stream()
-                                        .map(resourceMapper)
-                                        .map(ctpClient::execute)
-                                        .map(CompletionStage::toCompletableFuture);
+    public static <T extends Resource, C extends QueryDsl<T, C>, S> void queryAndCompose(
```

Comment:
- description/javadoc is missing
  - looks like comparing to `queryAndExecute()` this function doesn't compose anything additionally.

So, looks like either naming or documentation is missing/mismatching.

Reply:
Javadoc: e0239f6

Reply:
`queryAndExecute` - queries and executes a sphere request
`queryAndCompose` - queries and composes whatever stage.